src/main.ts

```diff
@@ -109,6 +150,27 @@ export const fromEither = <E, A>(either: Either<E, A>) => {
   return success<E, A>(either.right);
 };
 
+/**
+ * Construct a new `Dataway` from a `fp-ts` `Option` type, if the value is a `None`,
+ * returns `NotAsked` else returns the `Some` value wrapped in a `Success`
```

Comment:
```
Constructs a new `Dataway` from a `fp-ts` `Option` type. 
If the value is a `None` it returns `NotAsked`
otherwise it returns the `Some` value wrapped in a `Success`
```